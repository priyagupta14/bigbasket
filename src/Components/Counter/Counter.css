.product-counter {
  display: flex;
  justify-content: space-between;
  background-color: rgb(197, 190, 190);
  height: 3rem;
}
.product-counter > p {
  height: 1rem;
  width: 5rem;
}
.product-counter > button {
  width: 2rem;
}
