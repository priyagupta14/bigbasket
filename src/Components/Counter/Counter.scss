@import '../../styles/index.scss';

.product-counter {
  display: flex;
  justify-content: space-between;
  border: 1px solid black;
  background-color: $primary-background-color;
  height: 3rem;
}
.zeroProduct{
  background-color: rgb(148, 147, 147);
  display: flex;
  justify-content: space-between;
  border: 1px solid black;
  background-color: $primary-background-color;
  height: 3rem;
}
.moreProduct{
  display: flex;
  justify-content: space-between;
  border: 1px solid black;
  background-color: $primary-background-color;
  height: 3rem;
  background-color: rgb(211, 198, 123);
}
.product-counter > p {
  height: 1rem;
  width: 5rem;
  
}
.product-counter > button {
  width: 2rem;
  height: auto;
}
