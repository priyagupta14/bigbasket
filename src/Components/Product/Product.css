.product-card-element {
  border: solid black 1px;
  padding: 2rem;
  margin: 0.5rem;
}
.product-card-element > img {
  width: 10rem;
  height: 10rem;
}
