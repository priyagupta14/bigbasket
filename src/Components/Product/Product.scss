@import '../../styles/index.scss';

.product-card-element {
  border: solid black 1px;
  padding: 2rem;
  margin: 0.5rem;
  box-shadow: 10px 10px 5px rgb(209, 205, 205);
  border-radius: $border-radius;
  img{
    width: 10rem;
    height: 10rem;
  }
}
.price-counter{
  border-radius: 5px;
  padding: 0.5rem;
  background-color: rgb(207, 207, 207);
}
@media screen and (max-width: $breakpoint-for-phone) {
  .product-card{
    margin: auto;
    display: flex;
    justify-content: center;
  }
  h1{
    font-size: large;
  }
}