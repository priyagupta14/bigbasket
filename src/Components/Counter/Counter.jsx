import React, { Component } from "react";
import "./Counter.css";

export default class Counter extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div class="product-counter">
        <button onClick={this.props.onIncrement}>+</button>
        <p>{this.props.itemCount} in basket</p>
        <button onClick={this.props.onDecrement}>-</button>
      </div>
    );
  }
}
