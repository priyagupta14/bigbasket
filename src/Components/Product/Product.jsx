/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import Counter from '../Counter/Counter';
import './Product.scss';

const Product = (props) => (
  <div className="product-card-element">
    <img src={props.product.url} alt="fruitItem" />
    <p>{props.product.name}</p>
    <div className="price-counter">
      <p>
        MRP
        {props.product.price}
        /-
      </p>
      <Counter
        onIncrement={() => props.onIncrement(props.product.id, props.category)}
        onDecrement={() => props.onDecrement(props.product.id, props.category)}
        itemCount={props.product.inCartCount}
      />
    </div>
  </div>
);

Product.propTypes = {
  product: PropTypes.arrayOf(
    PropTypes.shape({
      price: PropTypes.number.isRequired,
      count: PropTypes.number.isRequired,
      url: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
  ).isRequired,
};
Product.propTypes = {
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};
export default Product;
