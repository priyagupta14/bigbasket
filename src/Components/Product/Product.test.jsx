/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import Product from './Product';
import Counter from '../Counter/Counter';

describe(Product.name, () => {
  const mockProps = {
    product:
      {
        id: 1,
        name: 'banana',
        url: 'banana',
        count: 0,
        price: 50,
      },

    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('sanity test', () => {
    expect(1 + 1).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(<Product {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
  it('should display product details', () => {
    render(<Product {...mockProps} />);
    screen.getByText('banana');
    // screen.getByTestId('banana');
    screen.getByText('MRP50/-');
    render(<Counter
      itemCount={mockProps.product.id}
      onDecrement={mockProps.onDecrement}
      onIncrement={mockProps.onIncrement}
    />);
  });
});
