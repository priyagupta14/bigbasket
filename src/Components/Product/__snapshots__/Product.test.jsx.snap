// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should match snapshot 1`] = `
<div>
  <div
    class="product-card-element"
  >
    <img
      alt="fruitItem"
      src="banana"
    />
    <p>
      banana
    </p>
    <div
      class="price-counter"
    >
      <p>
        MRP
        50
        /-
      </p>
      <div
        class="zeroProduct"
      >
        <button
          type="button"
        >
                  +      
        </button>
        <p>
          0
           
          in basket
        </p>
        <button
          type="button"
        >
          -
        </button>
      </div>
    </div>
  </div>
</div>
`;
